### PR TITLE
Unified widget variable naming

### PR DESCRIPTION
**Pull request by rakoenig**
Document widget abbrevations in ui-framework-documentation.md. 

The current state is that all "Page" code was revised and variables were renamed according to the ui-framework-documentation.md proposal.. 

- Related ticket: https://progress.opensuse.org/issues/105070
- Needles: n.a.
- Verification run: https://openqa.suse.de/tests/overview?distri=sle&version=15-SP4&build=rakoenig_widget_naming&groupid=96

### CODE REVIEW

File: ui-framework-documentation.md [hunk 1]
```diff
@@ -459,6 +459,30 @@ sub edit_proposal {
...
+
+### Abbreviated prefixes for libuyi widgets
+
+To avoid confusion identifiers that refer to UI widgets should be prefixed with a 2 to 4 letter abbreviation for the widget followed by an underscore '_'.
+
+| Prefix | Widget | Example     |
+|--------|--------|-------------|
+| btn    | Button | btn_ok      |
+| chb    | ChecBbox | chb_autologin |
+| cb     | ComboBox | cb_filesystem |
```
[the remark this answers is not in the document]
Comment (rakoenig) [reply ui-framework-documentation.md:471]: Good point George. When I wrote this I looked at what we  are using now, to minimize edits when we agree on a naming scheme. Switching to a scheme that is not applied at all at the moment would mean changinge practically everything in the code. Not that I'm afraid of that, I just wonder how many VRs we need for this.

[the remark this answers is not in the document]
Comment (ge0r) [reply ui-framework-documentation.md:471]: That's a valid point

**Review by ge0r — APPROVED** [review]
Very nice, I like the intuitive rules as well

File: ui-framework-documentation.md [hunk 2]
```diff
@@ -459,6 +459,39 @@ sub edit_proposal {
 
  edit_proposal({is_lvm => 1, has_separate_home => 1});
 ``` 
+
+### Abbreviated prefixes for libuyi widgets
```
Comment (jknphy) [inline ui-framework-documentation.md:463]: technically they are YuiRestClient Widgets (YuiRestClient::Widget) it is our construction to map libyui Widgets developed by YaST developers, libyui widget is for example: YRichText which lives here: https://github.com/libyui/libyui

File: ui-framework-documentation.md [hunk 3]
```diff
@@ -459,6 +459,39 @@ sub edit_proposal {
...
+
+To avoid confusion identifiers that refer to UI widgets should be prefixed with a 3  letter abbreviation for the widget followed by an underscore '_'.
```
Comment (jknphy) [inline ui-framework-documentation.md:465]: s/3  letter/3-letter/ or s/3  letter/three-letter/, right?

File: ui-framework-documentation.md [hunk 4]
```diff
@@ -459,6 +459,39 @@ sub edit_proposal {
...
+| lbl    | Label   | lbl_warning | 
+| mnc    | MenuCollection | mnc_operation |
+| prb    | ProgressBar | prb_total_packages |
+| rdb     | RadioButton | rdb_skip_registration |
+| rct     | RichText    | rct_welcome |
+| slb   | SelectionBox | slb_addons | 
+| tbl    | Table        | tbl_devices |
+| txb    | TextBox      | txb_maximum_channel | 
+| tre   | Tree         | tre_system_view |
+| tab    | Tab          | tab_boot_loader_settings |
```
Comment (jknphy) [inline ui-framework-documentation.md:491]: we could mention this special case as well on top with the rules.

**Review by jknphy — COMMENTED** [review]
LGTM, let's see what the other thinks and we can start to replace them.

File: ui-framework-documentation.md (hunk 4 above)
Comment (ge0r) [reply ui-framework-documentation.md:491]: doesn't this fall under the second rule?

File: ui-framework-documentation.md (hunk 3 above)
Comment (rakoenig) [reply ui-framework-documentation.md]: Agreed

**Review by ge0r — APPROVED** [review]
Rainer very nice work, I went through the changes and I think I didn't miss something, and VRs look good, so LGTM!

**Comment by jknphy** [comment]
we will need VR for maintenance for test suite using storage-ng.

**Review by jknphy — APPROVED** [review]
Definitely will help to catch inconsistencies.

**Comment by DimStar77** [comment]
Is that a possibility to cause https://openqa.opensuse.org/tests/2232091 ?

**Comment by jknphy** [comment]
@rakoenig it is taking a look, probably missing renaming when we exceptionally pass the widget as parameters:
```
lib/Installation/License/Sle/LicenseAgreementController.pm
lib/Installation/License/Sle/Firstboot/LicenseAgreementController.pm
```

**Pull request merged**